server/src/main/java/org/opentosca/toscana/plugins/kubernetes/docker/mapper/BaseImageMapper.java

```diff
@@ -66,9 +82,39 @@ Stores the update intervall for the base Image mappings (in hours)
     private MapperEngine engine;
 
     @Autowired
-    public BaseImageMapper(DockerBaseImages[] dockerBaseImages) {
+    public BaseImageMapper(DockerBaseImages[] dockerBaseImages, Preferences preferences) {
         this.baseImages = dockerBaseImages;
         engine = new MapperEngine(imageMap);
+        dataDir = new File(preferences.getDataDir(), DOCKER_IMAGE_DIR);
+        dataDir.mkdirs();
+        imageTagsFile = new File(dataDir, "image_tags");
+        lastUpdateFile = new File(dataDir, "last_update");
+        deserializeTags();
+
+    }
+
+    private void deserializeTags() {
+        try {
+            if (imageTagsFile.exists()) {
+                ObjectInputStream ois = new ObjectInputStream(new FileInputStream(imageTagsFile));
+                Object o = ois.readObject();
+                ois.close();
+                if (o instanceof List) {
+                    imageTags = (List<ImageTags>) o;
+                } else {
+                    logger.warn("File '{}' contained wrong data. Deleting file", imageTagsFile);
```

Comment:
You say something about deleting files but delete nothing?

Reply:
fixed